source/components/Autocomplete.js

```diff
@@ -187,29 +191,46 @@ class AutocompleteBase extends Component<AutocompleteProps, State> {
       maxSelections != null ? selectedOptions.length < maxSelections : true;
     const areFilteredOptionsAvailable =
       filteredOptions && filteredOptions.length > 0;
-
+    let skipValueSelection = false;
     if (
       !maxSelections ||
       (canMoreOptionsBeSelected && areFilteredOptionsAvailable)
     ) {
-      if (!selectedOption) return;
-      const option = selectedOption.trim();
-      const optionCanBeSelected =
-        (selectedOptions.indexOf(option) < 0 && !multipleSameSelections) ||
-        multipleSameSelections;
-
-      if (option && optionCanBeSelected && isOpen) {
-        const newSelectedOptions = _.concat(selectedOptions, option);
-        this.selectionChanged(newSelectedOptions, event);
-        this.setState({ selectedOptions: newSelectedOptions, isOpen: false });
+      if (!selectedOption || !selectedOption.length) return;
+      const option = _.isString(selectedOption) ? selectedOption.trim() : selectedOption;
+      let newSelectedOptions: Array<any> = [];
+      if (option && Array.isArray(option)) {
+        newSelectedOptions = [...selectedOptions];
+        option.forEach(item => {
+          const optionCanBeSelected = filteredOptions.includes(item) &&
+            !selectedOptions.includes(item) &&
+            !newSelectedOptions.includes(item);
```

Comment:
Isn't the `multipleSameSelections` missing here now?

Reply:
Yes i have updated now this part as we need to recheck multipleSameSelections && filteredOptions.includes(item) in both cases